### ISSUE TITLE
Problem with communication between server and agent

### ISSUE THREAD

**Issue by masterofoz** [PowerShellEmpire/Empire#54]
Hi!
I am planning to play around with this cool tool in the Offensive Security Labs.
I started to test this tool yesterday but ran into problems regarding the communication between the server and "client".
I have generated the .bat file stager and launched it on a Windows 7 box.
I already have the server started on my Kali Linux box (no firewall) and I have a listener started as well.

After I run the stager on the Windows 7 box it connects to my listener/server and all looks fine.
The problem is that the only command I can run that works is "info" when I try to interact with the client. Also, I can not see any useful info in the client folder on the server.

(Empire: agents) > interact MWVMCKCYUSSAZBEH
(Empire: MWVMCKCYUSSAZBEH) > 
(Empire: MWVMCKCYUSSAZBEH) > info

Every other command I try to run results in no response now matter how long I wait.
The client status looks fine, it is connected and looks ok as far as I can see.
I have tried to restart the server and run another client session but the result is the same, I have also turned off the Win 7 firewall in case it was causing any problems, but no difference.

When I tried to troubleshoot the problem I could see this:
-If I run netstat on my Win 7 box I can't see any connection between the client and the server (Kali Linux machine)
-I have started Empire in debug mode and looked at the logs and it looks like the client is asking for status on a regular basis(?), but the client does not respond to any commands/tasks.
It looks like this in the logs:
2015-09-28 11:59:05 HttpHandler : [*] /news.asp requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:11 HttpHandler : [*] /login/process.jsp requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:13 Agents : [*] Tasked MWVMCKCYUSSAZBEH to run TASK_SHELL
2015-09-28 11:59:16 HttpHandler : [*] /admin/get.php requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:21 HttpHandler : [*] /login/process.jsp requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:27 HttpHandler : [*] /admin/get.php requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:32 HttpHandler : [*] /login/process.jsp requested from MWVMCKCYUSSAZBEH at 192.168.31.118
2015-09-28 11:59:37 HttpHandler : [*] /login/process.jsp requested from MWVMCKCYUSSAZBEH at 192.168.31.118

[*] Active agents:

  Name               Internal IP     Machine Name    Username            Process             Delay    Last Seen
  ---------          -----------     ------------    ---------           -------             -----    --------------------
  MWVMCKCYUSSAZBEH   192.168.31.118  OFFSEC-LAB      *offsec-lab\offsec  powershell/2640     5/0.0    2015-09-28 08:24:37

This is the agent info:
[*] Agent info:

	ps_version      	2
	old_uris        	None
	jitter          	0.0
	servers         	None
	internal_ip     	192.168.31.118
	working_hours   	
	session_key     	4d[}Y<MN$Bv:xIG=LSPH56\)(Tif1`0X
	children        	None
	checkin_time    	2015-09-28 07:10:54
	hostname        	OFFSEC-LAB
	delay           	5
	uris            	/admin/get.php,/news.asp,/login/process.jsp
	username        	offsec-lab\offsec
	kill_date       	
	parent          	None
	process_name    	powershell
	listener        	http://192.168.30.89:8080/
	sessionID       	MWVMCKCYUSSAZBEH
	process_id      	2640
	os_details      	Microsoft Windows 7 Professional N
	lost_limit      	60
	ID              	1
	name            	MWVMCKCYUSSAZBEH
	external_ip     	192.168.31.118
	headers         	
	user_agent      	Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
	lastseen_time   	2015-09-28 08:25:46
	high_integrity  	1

Do you know what could be the problem?

**Comment by enigma0x3** [comment]
Can you provide a little info about your environment? Is the machine on a domain and some system settings, such as what language pack you are using?

**Comment by butterworth1492** [comment]
I have the same issue.  

- I am not in a Windows domain

- English

- I've tried the vanilla launcher as well as the launcher.bat

- I'm running the stager on a Windows machine that is calling back to a guest Kali 1.1 VM running on that same box

- I am also unable to kill agents after they've called back.  I've specified the particular agent as well as "all".  The agents persist through Empire restart as well as killing the powershell process on the target box.

**Comment by enigma0x3** [comment]
agents will remain in the menu, even if inactive. You can remove them by typing "remove stale" or "remove <agentName>". Are you also unable to run commands against live agents?

**Comment by butterworth1492** [comment]
Correct.  I have the same issue as OP.  I get a fresh callback and it is unresponsive.

**Comment by butterworth1492** [comment]
Not much to see here.  Let me know if I've fat-fingered something or failed to RTFM.  

![listener](https://cloud.githubusercontent.com/assets/8503235/10249526/d08e455e-68f3-11e5-92bc-17729017413e.png)
![stager](https://cloud.githubusercontent.com/assets/8503235/10249527/d3c2d186-68f3-11e5-94cd-f98ce3052681.png)
![connection](https://cloud.githubusercontent.com/assets/8503235/10249532/d76a16fa-68f3-11e5-9201-5b137f56d739.png)
![agent](https://cloud.githubusercontent.com/assets/8503235/10249534/da481ef8-68f3-11e5-897c-63c36734636c.png)

**Comment by enigma0x3** [comment]
I noticed that you changed the DefaultProfile on your listener. Did you create one? That option is for setting a profile to make traffic over the wire to look like something else. What you set looks like an attempt to alter the URI. For example, you can see some already configured profiles in the /data/profiles folder. Just copy out the string (without the quotes) and set that as the DefaultProfile to change it. 

As a quick check, can you go into empire/setup and run reset.sh? This will set everything back to the defaults. Give that a try and see if you are able to execute commands.

**Comment by enigma0x3** [comment]
@masterofoz, can you paste in your listener configuration?

**Comment by butterworth1492** [comment]
Yes...I had misunderstood the DefaultProfile option.  Thanks for clearing that up.  Unfortunately, I'm still having the problem.  :(

Having run reset.sh, here is the list of commands I'm running from a fresh Empire prompt...
-------------------------------------------------------
listeners
set Name HttpListener
set Host http://192.168.72.135
set Port 8080
execute
main
usestager launcher_bat
set Listener HttpListener
set OutFile /opt/Empire/test/launcher.bat
generate
main
-------------------------------------------------------

I then put the batch file on the target and run it from cmd.exe.  Callback, 'interact <agent_id>', 'sysinfo', nothing.

I'm probably missing something simple, here.

**Comment by enigma0x3** [comment]
Are you running under a English - US language setup on the target?

**Comment by butterworth1492** [comment]
Yes I am.

**Comment by ronnieflip** [comment]
Has anyone found a work around this problem?

**Comment by enigma0x3** [comment]
Just curious, when was the last time you did a fresh clone/pull?

**Comment by ronnieflip** [comment]
Did my last pull two days ago.

**Comment by butterworth1492** [comment]
I cloned yesterday.  Haven't done a fresh pull since.

**Comment by enigma0x3** [comment]
Can you start up wireshark on your target box and see what the traffic looks like when a command is sent to the target? If you can, post some of the details of the cap back here.

**Comment by butterworth1492** [comment]
Here's some basic stuff.

Here's ambient traffic without my touching the keyboard...
![ambient](https://cloud.githubusercontent.com/assets/8503235/10255241/1c51d5aa-6917-11e5-9433-a35686006ba5.png)

Here's the stream for a beacon when no task has been requested...
![content](https://cloud.githubusercontent.com/assets/8503235/10255244/1eb9c32a-6917-11e5-8462-0946accf7ced.png)

Here's the stream for a beacon when a task has been requested...
![content2](https://cloud.githubusercontent.com/assets/8503235/10255248/21bbd3ce-6917-11e5-9c80-fe1e294e6bb9.png)

All I was doing for this test was running 'sysinfo' over and over.

**Comment by butterworth1492** [comment]
Forgot to mention...when running the sysinfo command, the two following HTTP GET sessions saw no change in periodicity or content.

- Ran 'sysinfo' in Empire
- Next GET request from target had encrypted response from Empire.  No additional traffic generated.
- Next traffic was ~7 seconds later.  Traffic was unencrypted "It works!" message with no additional traffic.

(Hope this is helping.)

**Comment by masterofoz** [comment]
Hi!

Thanks for the response!

None of my hosts (Kali Linux or Windows 7 Box) are in a domain, both are english version.
However, I usually change the language profile on my Kali Linux box to swedish for my keyboard to work as I like.

Another thing worth mentioning is that my Kali Linux Box has two ip adresses, the first (eth0) is the one that is connected to my home network and Internet, the second (tap0) is my VPN adresses which I use to access the Offensive Security Lab Network.

So I have changed the automatically choosen listener adress in the settings for the listerner to the VPN adress (below):
Host              True        http://192.168.30.89:8080        Hostname/IP for staging.

Thats the same adress I use for all my activity in the lab network (but when I run Empire I will usually let it run alone).

This is my settings for the listerner:

(Empire: listeners) > set Host http://192.168.30.89:8080
(Empire: listeners) > info

Listener Options:

  Name              Required    Value                            Description
  ----              --------    -------                          -----------
  KillDate          False                                        Date for the listener to exit (MM/dd/yyyy).
  Name              True        test                             Listener name.
  DefaultLostLimit  True        60                               Number of missed checkins before exiting
  StagingKey        True        !)~H_L|<^EFTAyfGIU/;`\DS[=K.b{&7 Staging key for initial agent negotiation.
  Type              True        native                           Listener type (native, pivot, hop, foreign, meter).
  RedirectTarget    False                                        Listener target to redirect to for pivot/hop.
  DefaultDelay      True        5                                Agent delay/reach back interval (in seconds).
  WorkingHours      False                                        Hours for the agent to operate (09:00-17:00).
  Host              True        http://192.168.30.89:8080        Hostname/IP for staging.
  CertPath          False                                        Certificate path for https listeners.
  DefaultJitter     True        0.0                              Jitter in agent reachback interval (0.0-1.0).
  DefaultProfile    True        /admin/get.php,/news.asp,/login/ Default communication profile for the agent.
                                process.jsp|Mozilla/5.0 (Windows
                                NT 6.1; WOW64; Trident/7.0;
                                rv:11.0) like Gecko
  Port              True        8080                             Port for the listener.

(Empire) > listeners

[*] Active listeners:

  ID    Name              Host                                 Type      Delay/Jitter   KillDate    Redirect Target
  --    ----              ----                                 -------   ------------   --------    ---------------
  1     test              http://192.168.30.89:8080            native    5/0.0  


After I changed the listerner adress I generate a launcher.bat with this command:

(Empire: listeners) > usestager launcher_bat test
(Empire: stager/launcher_bat) > execute

[*] Stager output written out to: /tmp/launcher.bat

I did as you suggested and ran the reset.sh, all the info above from the listerner is after I have started Empire after resetting it.

This is the client info, both machines are virtual machines running on the VMWare platform. 
I run VMWare Fusion on my Mac and I guess Offsec are running the enterprise version of VMWare.
I run the Empire bat-file (client/agent) as Administrator on the Windows Box so it is likely not a permission issue.

C:\Users\Administrator>systeminfo

Host Name:                 OFFSEC-LAB
OS Name:                   Microsoft Windows 7 Professional N
OS Version:                6.1.7601 Service Pack 1 Build 7601
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Offsec
Registered Organization:
Product ID:                00376-827-5000002-85111
Original Install Date:     4/8/2013, 2:45:11 AM
System Boot Time:          9/30/2015, 9:31:48 PM
System Manufacturer:       VMware, Inc.
System Model:              VMware Virtual Platform
System Type:               X86-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: x64 Family 6 Model 44 Stepping 2 GenuineIntel ~
3458 Mhz
BIOS Version:              Phoenix Technologies LTD 6.00, 6/22/2012
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-05:00) Eastern Time (US & Canada)
Total Physical Memory:     1,024 MB
Available Physical Memory: 682 MB
Virtual Memory: Max Size:  2,048 MB
Virtual Memory: Available: 1,736 MB
Virtual Memory: In Use:    312 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\OFFSEC-LAB
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB2534111
                           [02]: KB2621440
                           [03]: KB2667402
                           [04]: KB976902
Network Card(s):           1 NIC(s) Installed.
                           [01]: Intel(R) PRO/1000 MT Network Connection
                                 Connection Name: offsec
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.31.118

I run the newly generated launcher.bat and the agent connetcs to the server:

(Empire: stager/launcher_bat) > agents
[!] No agents currently registered 

(Empire: agents) > [+] Initial agent N213UNPUNE2G4SZK from 192.168.31.118 now active

(Empire: agents) > interact N213UNPUNE2G4SZK

(Empire: N213UNPUNE2G4SZK) > info

[*] Agent info:

	ps_version      	2
	old_uris        	None
	jitter          	0.0
	servers         	None
	internal_ip     	192.168.31.118
	working_hours   	
	session_key     	^Oeg%QRX(p5o.83TMj}x9bl:$u=m`YSU
	children        	None
	checkin_time    	2015-10-04 06:18:05
	hostname        	OFFSEC-LAB
	delay           	5
	uris            	/admin/get.php,/news.asp,/login/process.jsp
	username        	offsec-lab\offsec
	kill_date       	
	parent          	None
	process_name    	powershell
	listener        	http://192.168.30.89:8080/
	sessionID       	N213UNPUNE2G4SZK
	process_id      	1812
	os_details      	Microsoft Windows 7 Professional N
	lost_limit      	60
	ID              	1
	name            	N213UNPUNE2G4SZK
	external_ip     	192.168.31.118
	headers         	
	user_agent      	Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
	lastseen_time   	2015-10-04 06:18:44
	high_integrity  	1

This is basically the conversation I see between the two clients in Wireshark (running Wireshark on the Kali Linux box):

GET /login/process.jsp HTTP/1.1
Cookie: SESSIONID=N213UNPUNE2G4SZK
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
Host: 192.168.30.89:8080
Connection: Keep-Alive

HTTP/1.0 200 OK
Server: Microsoft-IIS/7.5 
Date: Sun, 04 Oct 2015 10:24:05 GMT

<html><body><h1>It works!</h1><p>This is the default web page for this server.</p><p>The web server software is running but no content has been added, yet.</p></body></html>

and

GET /news.asp HTTP/1.1
Cookie: SESSIONID=N213UNPUNE2G4SZK
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
Host: 192.168.30.89:8080
Connection: Keep-Alive

HTTP/1.0 200 OK
Server: Microsoft-IIS/7.5 
Date: Sun, 04 Oct 2015 10:24:00 GMT

v..;..1&........u..>%x....n.T.$...).D....@..y.....cR

and

GET /admin/get.php HTTP/1.1
Cookie: SESSIONID=N213UNPUNE2G4SZK
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
Host: 192.168.30.89:8080
Connection: Keep-Alive

HTTP/1.0 200 OK
Server: Microsoft-IIS/7.5 
Date: Sun, 04 Oct 2015 10:23:49 GMT

<html><body><h1>It works!</h1><p>This is the default web page for this server.</p><p>The web server software is running but no content has been added, yet.</p></body></html>

Unfortunatelly when I try to interact with the client it is not responding to my commands:

(Empire: N213UNPUNE2G4SZK) > ls
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) > sysinfo
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) > sysinfo
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) > 
(Empire: N213UNPUNE2G4SZK) >

**Comment by masterofoz** [comment]
I have ran Wireshark on the Win 7 Box while i ran "ls" and "sysinfo" commands.
I have cant attach a filtered view of the capture (filtered on http) with the raw data from the packets since GitHub is not allowing it, but I can mail it if it would be useful.

The traffic looks like this:
"No.","Time","Source","Destination","Protocol","Length","Info"
"1","0.000000000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"2","0.163737000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"3","0.164018000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"4","5.334072000","192.168.31.118","192.168.30.89","HTTP","253","GET /admin/get.php HTTP/1.1 "
"5","5.498498000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"6","5.498776000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"7","10.685192000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"8","10.849932000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"9","10.850322000","192.168.30.89","192.168.31.118","HTTP","173","HTTP/1.0 200 OK "
"10","16.036607000","192.168.31.118","192.168.30.89","HTTP","253","GET /admin/get.php HTTP/1.1 "
"11","16.200617000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"12","16.200882000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"13","21.372243000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"14","21.536840000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"15","21.537088000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"16","26.722288000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"17","26.887058000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"18","26.887122000","192.168.30.89","192.168.31.118","HTTP","173","HTTP/1.0 200 OK "
"19","32.105061000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"20","32.269711000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"21","32.270090000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"22","37.439552000","192.168.31.118","192.168.30.89","HTTP","257","GET /login/process.jsp HTTP/1.1 "
"23","37.603996000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"24","37.604239000","192.168.30.89","192.168.31.118","HTTP","173","HTTP/1.0 200 OK "
"25","42.805189000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"26","42.969981000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"27","42.970186000","192.168.30.89","192.168.31.118","HTTP","173","HTTP/1.0 200 OK "
"28","48.140566000","192.168.31.118","192.168.30.89","HTTP","248","GET /news.asp HTTP/1.1 "
"29","48.305286000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"30","48.305653000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"31","53.494630000","192.168.31.118","192.168.30.89","HTTP","253","GET /admin/get.php HTTP/1.1 "
"32","53.658922000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"33","53.659194000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "
"34","58.873731000","192.168.31.118","192.168.30.89","HTTP","257","GET /login/process.jsp HTTP/1.1 "
"35","59.038932000","192.168.30.89","192.168.31.118","TCP","71","[TCP segment of a reassembled PDU]"
"36","59.039218000","192.168.30.89","192.168.31.118","HTTP","294","HTTP/1.0 200 OK "

Basically all I see is "It Works" messages on a regular basis.

**Comment by masterofoz** [comment]
Today I tried to run Empire against one of the lab machines that we are supposed to hack in the course.
I got exactly the same error (basic communication between server and agent works but I can't run any commands on the client), I guess this may indicate that the problem is on the server side?

The machine with the client agent has this configuration:

(Empire: RGRTTAP1R2BBHZ2K) > info
[*] Agent info:

	ps_version      	4
	old_uris        	None
	jitter          	0.0
	servers         	None
	internal_ip     	192.168.31.250
	working_hours   	
	session_key     	Ps4H>uN&}58~p$?JTFhA+3(MOQ%#Z|W:
	children        	None
	checkin_time    	2015-10-04 14:19:42
	hostname        	BETHANY
	delay           	5
	uris            	/admin/get.php,/news.asp,/login/process.jsp
	username        	Bethany\Bethany
	kill_date       	
	parent          	None
	process_name    	powershell
	listener        	http://192.168.30.89:8080/
	sessionID       	RGRTTAP1R2BBHZ2K
	process_id      	3024
	os_details      	Microsoft Windows 8.1 Enterprise
	lost_limit      	60
	ID              	2
	name            	RGRTTAP1R2BBHZ2K
	external_ip     	192.168.31.250
	headers         	
	user_agent      	Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
	lastseen_time   	2015-10-04 14:21:39
	high_integrity  	0

I am able to run another powershell tool on this machine, powercat (a powershell version of netcat), and that tool works as expected, but I guess it is less complicated and have only basic functions as netcat have. It is also started by a bat script and connects back to my netcat listener on the Kali Linux host.

**Comment by enigma0x3** [comment]
hmmmm...thanks for the info guys! Let me ping @sixdub and @harmj0y and we will get back to you guys. 

One thing to try is to see if something in master broke this for you. When you can, can you checkout an older tagged release and try that? This will help us narrow it down to either a problem with Empire or something in the environment that is interrupting normal Empire functionality.

**Comment by enigma0x3** [comment]
Hey all,
Just checking to see if anyone having this issues has had a chance to checkout a previous tag and see if the issue exists. I would recommend checking out the 1.0 tag and testing. If it works in 1.0, move up the tag chain until it breaks. That will give us an idea what release broke this for you. 

If you are unfamiliar with checking out a tag, you can do a "git checkout <doubleTab>" to view options. You can then do a "git checkout <option" to check it out...so "git checkout 1.0" to checkout tag 1.0. 

Let me know what you find.

**Comment by butterworth1492** [comment]
Hey.  I started with 1.0 and am still having the same issue.

(Ignore the listener error...it bound to the port as it was supposed to.)

![empire_1 0_error](https://cloud.githubusercontent.com/assets/8503235/10490806/0422e6aa-7271-11e5-87f4-0096c93e8ed3.png)

**Comment by butterworth1492** [comment]
FYI...everything I've done has been in Kali 1.1 so far.  Running in Kali 2 this morning seems to work fine.

Let me know if you need any package deltas between my two instances.

**Comment by enigma0x3** [comment]
@butterworth1492,
I'm going to pull down a fresh Kali 1.1 iso and test this. If you can, can you share package deltas between the two instances?

Thanks for your help and patience during this. I hope to have this figured out soon!

**Comment by enigma0x3** [comment]
So I pulled the 64bit iso from here: http://cdimage.kali.org/kali-1.1.0/, updated it, installed vmware tools, did a fresh clone of https://github.com/powershellempire/empire.git, ran install.sh in the setup folder and then started Empire up. Running the stager on a windows 7 x64 box gives me a callback and the ability to run commands without a problem.

Can you maybe try grabbing a fresh Kali 1.1 install, updating it and then trying Empire? I'm curious if you have a package or something that interferes with something Empire uses for comms.

**Comment by butterworth1492** [comment]
No errors or feedback from the "-w Hidden" suggestion. Business as usual.

**Comment by enigma0x3** [comment]
Interesting. Give a frest Kali 1.1 install a try and let me know.

I'm running out of things to test :)

**Comment by ronnieflip** [comment]
Just done a fresh install of kali 1.1 but still facing the same challenge.

**Comment by enigma0x3** [comment]
@butterworth1492, but Empire works on Kali 2.0 fine?

**Comment by butterworth1492** [comment]
Correct.

**Comment by enigma0x3** [comment]
That is really odd (and interesting). I honestly have no idea what would cause that (other than package conflicts). Empire was built on Kali 1.x, so I'm really surprised to hear that it works on 2.0 and not 1.x. 

@ronnieflip, have you tried 2.0? I'm curious if what you guys are seeing is related.

**Comment by lolzitsgreg** [comment]
I was having very similar issues, but discovered that correcting the time on my server (Kali 2.0) fixed it.

**Comment by enigma0x3** [comment]
Interesting...can anyone with this issue on Kali 1.x try checking the time?

**Comment by masterofoz** [comment]
I haven't checked, but I know that I run the default time zone on my Kali Linux 1.x box.
That is, I have never changed timezone to my local time zone so it is running NY, US Time.
I am not sure when I write this what time zone the PWK Labs are in, but it might be US.
I am however located in Sweden, if there somehow is a time sync issue. 
My Mac that I run the Kali VM on is set to the Swedish time zone.

**Comment by lolzitsgreg** [comment]
I'm not sure that time zone would matter (and I haven't actually looked through the code for the PSE server and client), but if Kerberos is used to auth between the two, the clocks probably can't be more than 5 minutes apart (reference: https://en.wikipedia.org/wiki/Kerberos_(protocol)#Drawbacks_and_limitations).

**Comment by HarmJ0y** [comment]
I'm guessing the likely issue is the epoch syncing- on one of the client negotiation steps, [the server sends its current epoch time](https://github.com/PowerShellEmpire/Empire/blob/master/lib/common/agents.py#L1206-L1225) to the agent the agent uses to calculate the [epoch diff between it and the C2 server](https://github.com/PowerShellEmpire/Empire/blob/master/data/agent/agent.ps1#L149). This is used with a 5 minute sliding window [to validate packets on the agent side](https://github.com/PowerShellEmpire/Empire/blob/master/data/agent/agent.ps1#L867-L869) as well as the [packets on the server side](https://github.com/PowerShellEmpire/Empire/blob/master/lib/common/packets.py#L89).

The purpose of this was to provide some anti-replay functionality. My guess is that the time components are getting botched somehow. If you comment out [this code in agent.ps1](https://github.com/PowerShellEmpire/Empire/blob/master/data/agent/agent.ps1#L867-L869) and replace [this line in packets.py](https://github.com/PowerShellEmpire/Empire/blob/master/lib/common/packets.py#L89) with "return True", that should side step the epoch validation.

**Comment by enigma0x3** [comment]
Hey guys,
Has anyone had the chance to check if epoch might be causing this?

Thanks!

**Comment by ronnieflip** [comment]
Am using kali 2.0 but correcting the time settings did the job for me.

Thanks guys

**Comment by enigma0x3** [comment]
Yea, sounds like an epoch issue. Correcting the time should fix it.

Thanks guys!

**Comment by necony286** [comment]
THX i was thinking i was never gonna fix this but nice one @HarmJ0y  
it was the epoch issue. 
i did what the man said and i got it working :D 
(kali2.0 64bit)

**Comment by limingke** [comment]
thanks  终于 解决了这个困扰我很久的问题。。。谢谢大家。。。

**Comment by bayp1p0** [comment]
3.2

**Comment by bayp1p0** [comment]
Hı ı have some error on parrot 3.2.  When ı use my wan ıp with noıp connection work but ı can do nothıng ın interact :(
